obd_ontology/knowledge_graph_query_tool.py

```diff
@@ -964,6 +964,33 @@ def query_sub_components_by_component(self, component_name: str, verbose: bool =
             """
         return [row['sub_comp_name']['value'] for row in self.fuseki_connection.query_knowledge_graph(s, False)]
 
+    def query_super_component(self, sub_component: str, verbose: bool = True) -> List[str]:
+        """
+        Queries the super component for the specified sub component.
+
+        :param sub_component: sub component to query super component for
+        :param verbose: if true, logging is activated
+        :return: super component
+        """
+        if verbose:
+            print("########################################################################")
+            print(colored("QUERY: super component by sub component " + sub_component, "green", "on_grey", ["bold"]))
+            print("########################################################################")
+        comp_entry = self.complete_ontology_entry('SuspectComponent')
+        name_entry = self.complete_ontology_entry('component_name')
+        sub_comp_entry = self.complete_ontology_entry('SubComponent')
+        element_of_entry = self.complete_ontology_entry('elementOf')
+        s = f"""
+            SELECT ?comp_name WHERE {{
+                ?sub_comp a {sub_comp_entry} .
+                ?sub_comp {name_entry} "{sub_component}" .
+                ?comp a {comp_entry} .
+                ?comp {name_entry} ?comp_name .
+                ?sub_comp {element_of_entry} ?comp .
+            }}
+            """
+        return [row['comp_name']['value'] for row in self.fuseki_connection.query_knowledge_graph(s, False)]
```

Comment:
Why not pass verbose to the verbose parameter of query_knowledge_graph?

Reply:
I think I only wanted to control the string at the beginning of the function, but you're right, it's inconsistent. I fixed it.